### PR TITLE
Book model test

### PR DESCRIPTION
**Pull request by GabrielPiccini**
## Summary

Test for book model successfully implemented and previous card bugs fixed.
- [ ] Added the ".env" file with the secret key.
- [ ] Changed the jwt.secret content to look into secrets file to find the secret key
- [ ] Gems installed and configured:  faker, rspec-rails, factory_bot_rails, shoulda-matchers and database_cleaner-active_record.
- [ ] Created the books factory with random info given by faker.
- [ ] Tests for book model successfully implemented.

## Screenshots

image

## TRELLO Card

https://trello.com/c/ZHl5bePC/23-crear-tests-para-el-modelo-de-book

### CODE REVIEW

File: spec/factories/books.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :book do
+    genre { Faker::Book.genre }
+    author { Faker::Book.author }
+    title { Faker::Book.title }
+    publisher { Faker::Book.publisher }
+    image { Faker::Avatar.image }
+    year { Faker::Date.between(from: '1900-09-23', to: '2020-09-25') }
```
Comment (melissaregnier) [inline spec/factories/books.rb]: Take into consideration that this returns a Date, rather than a string.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+describe Book, type: :model do
+  it do
+    should validate_presence_of(:genre)
+  end
+  it do
```
Comment (melissaregnier) [inline spec/models/book_spec.rb:8]: add a newline between `it` blocks

File: spec/models/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    should validate_presence_of(:year)
+  end
+
+  subject(:book) do
+    build(:book)
+  end
+
+  it do
+    is_expected.to be_valid
+  end
```
Comment (melissaregnier) [inline spec/models/book_spec.rb:34]: add a newline after this

File: spec/models/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  describe '#create' do
+    context 'When the author is nil' do
+      subject(:book) do
+        build(:book, author: nil)
+      end
+
+      it do
+        is_expected.to be_invalid
+      end
+    end
```
Comment (melissaregnier) [inline spec/models/book_spec.rb:45]: add a newline between `context` blocks

**Review by melissaregnier — CHANGES_REQUESTED** [review]

**Review by dante-wd — APPROVED** [review]

**Pull request merged**